web/src/components/inventory/ShowNearby.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import useNuiEvent from '../../hooks/useNuiEvent';
+import { giveTo } from '../../dnd/onGive';
+import { useExitListener } from '../../hooks/useExitListener';
+
+interface Player {
+  player: number;
+}
+
+const ShowNearby: React.FC<{ isInventoryOpen: boolean }> = ({ isInventoryOpen }) => {
+  const [nearbyVisible, setNearbyVisible] = useState(false);
+  const [nearbyPlayers, setNearbyPlayers] = useState([]);
+  const [currentSlot, setSlot] = useState(0);
+  const [currentAmount, setAmount] = useState(0);
+
+  useNuiEvent('showNearby', (data) => {
+    if (nearbyVisible) {
```

Comment:
You should be closing inventory on dialog open, looks awkward when both are open at the same time.

Reply:
I'm not sure how to do it without messing up the structure of the codes. I'm thinking of doing this with players able to close the dialog when clicking outside the dialog. Not sure where to place the code, maybe in the hooks? and honestly not too confident if mine will be a good code haha.